chains/evm/cli/bridge/register-generic-resource.go

```diff
@@ -103,19 +103,46 @@ func ProcessRegisterGenericResourceFlags(cmd *cobra.Command, args []string) erro
 	ResourceIdBytesArr = callsUtil.SliceTo32Bytes(resourceIdBytes)
 
 	if Hash {
+		// We must check whether both a deposit and execute function signature is provide or else
+		// an invalid hash of 0x00000000 will be taken and set as a function selector in the handler
 		DepositSigBytes = callsUtil.GetSolidityFunctionSig([]byte(Deposit))
 		ExecuteSigBytes = callsUtil.GetSolidityFunctionSig([]byte(Execute))
+		if Deposit == "00000000" {
+			DepositSigBytes = [4]byte{}
+		}
+		if Execute == "00000000" {
+			ExecuteSigBytes = [4]byte{}
+		}
 	} else {
-		copy(DepositSigBytes[:], []byte(Deposit)[:])
-		copy(ExecuteSigBytes[:], []byte(Execute)[:])
+		depositBytes, err := hex.DecodeString(Deposit)
```

Comment:
You could probably use the DepositSigBytes variable here, and then omit the copy stuff

Reply:
Do you mean substituting `depositBytes` with `DepositSigBytes` from the `hex.DecodeString` call? Currently `DepositSigBytes` is type `[4]byte` whereas `DecodeString` returns `[]byte` and can be used to assign a value to `DepositSigBytes`. This could probably still be done but would require slight changes to all other places that use `DepositSigBytes`